### ISSUE TITLE
Block more classes from polymorphic deserialization (CVE-2018-14718 - CVE-2018-14721)

### ISSUE THREAD

**Issue by cowtowncoder** [FasterXML/jackson-databind#2097]
This issue covers following CVEs related to polymorphic deserialization, gadgets:

* CVE-2018-14718: RCE with slf4j-ext jar
* CVE-2018-14719: RCE with blaze-ds-opt, -core jars
* CVE-2018-14720: exfiltration/XXE with only JDK classes (some JDK versions)
* CVE-2018-14721: exfiltration/SSRF with axis2-jaxws

**Comment by bbossola** [comment]
When is the release of 2.8.11.3, with this fix, planned?

**Comment by bbossola** [comment]
bump :)

**Comment by cowtowncoder** [comment]
@bbossola I don't know. Question here is whether I'd release it now, or wait for another 4 classes that I know need to be blocked. Given there are 100+ dependencies for 2.8.11.2 there does seem to be demand for patches (which is sort of positive surprise).

**Comment by cowtowncoder** [comment]
Fix released on 23-Nov-2018, in:

* 2.7.9.5 (micro-patch of `jackson-databind`)
* 2.8.11.3 (micro-patch of `jackson-databind`, plus `jackson-bom` version `2.8.11.20181123` )

and will be included in `2.9.8` as soon as that gets released (full release along with other fixes)

**Comment by cedricdangremont** [comment]
@cowtowncoder.
when checking the archive for  2.6.7.2, I can see the content of the commit 87d29af25e82a249ea15858e2d4ecbf64091db44 which fixes the issue. Anyways, when I check the next releases (2.7.9.5, 2.8.11.3, 2.9.8), I don't find the associated code. In these versions the complete static initializer part containing the fix is missing.
Are these versions really fixed?
Thanks in advance.

**Comment by cedricdangremont** [comment]
@cowtowncoder please forget about my previous question, I have finally found out that the fix in the later releases was impacting another file. Please apologize.

Thanks
Cedric

**Comment by cowtowncoder** [comment]
@cedricdangremont no need to apologize, tracking this is not as easy as it should be. Glad you figured it out.

**Comment by cowtowncoder** [comment]
@eyecats Probably someone just read "before 2.9.8" to mean literally all versions, and is not familiar with the concept of multiple open release branches or something.

To me CVE system is much more hassle than value at this point: not only is there no concept of "MAY apply depending on your setting/usage" but also there is no useful way to manage or set versions after the fact (or maybe there is and I just have no resources to learn how to manage them -- or actual time to do that if I did). In this case I had an idea of version number for fixed-in, wrt 2.9, but in general fixed-in can only be known in hindsight, long after initial submission.
If anyone wants to submit additional information feel free to.

Worse, there are so many new tools that are being pushed in name of security that are based on flawed data, leading to unnecessary work. I realize that tooling would be useful if it reduced cognitive load, but unfortunately often they seem to over-simplify things into useless category... "as simple as possible but no simpler" as mr. Einstein stated it.
What a huge time sink, for absolutely no value for maintainers.

Anyway: I can't offer more information than to say that those 4 CVEs (... that I filed initially...)
